PriceCalculatorKata/DiscountCalculator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using PriceCalculatorKata.Common;
+
+namespace PriceCalculatorKata;
+
+public static class DiscountCalculator
+{
+    private static int _discountPercentage;
+    public static int DiscountPercentage
+    {
+        get => _discountPercentage;
+        set
+        {
+            if (!value.IsValid())
+                throw new ArgumentException("Invalid Discount Amount!", $"{value}");
+            _discountPercentage = value;
+        }
+    }
+
+    public static Dictionary<string, int> SpecialDiscounts = new();
+
+    public static float CalculateDiscountAmount(float price)
+    {
+        if (DiscountPercentage <= 0) return 0;
+        var discount = DiscountPercentage / 100F;
+        var discountAmount = price * discount;
+        return discountAmount;
+    }
+
+    public static void AddSpecialDiscount(string productCode, int specialDiscountPercentage)
+    {
+        SpecialDiscounts.Add(productCode, specialDiscountPercentage);
+    }
+    
+    public static bool SpecialDiscountExists(Product product)
+    {
+        var productDiscountsExists = SpecialDiscounts.ContainsKey(product.UniversalProductCode);
+        if (SpecialDiscounts.Any() && productDiscountsExists) return true;
+        return false;
+    }
+
+    public static float CalculateSpecialDiscountAmount(Product product)
+    {
+        if (!SpecialDiscountExists(product)) return 0;
+        var specialDiscount = SpecialDiscounts[product.UniversalProductCode] / 100F;
```

Comment:
The use of Dictionary.TryGetValue her will get rid of the need for SpecialDiscountExists method

Reply:
> The use of Dictionary.TryGetValue her will get rid of the need for SpecialDiscountExists method

I have removed the Exists method and swapped it with Dictionary.TryGetValue. Thank you for the suggestion!